ZichanYang324/BF-Employee-HR
# BGP-6[Frontend and Backend] Link onboarding application frontend to backend

# Description
Implements BGP-6, link onboarding application page frontend to backend

## Main changes explained:
- Sending text data and file data together with `FormData`
- Backend logic for receiving text and file data correctly
…

## How to test:
1. check into current branch
2. Submit an onboarding application on frontend
3. Check data base and console of backend

## Screenshots or videos of changes or expecting result:
New profile created in DB

![image](https://github.com/ZichanYang324/BF-Employee-HR/assets/85213835/f510ffed-3644-454c-a766-ad3fd3d0f3ef)

and received files printed in backend console

![image](https://github.com/ZichanYang324/BF-Employee-HR/assets/85213835/758c7d69-913c-47b8-89b8-d64492bfd923)

## Bug
can not read properties of undefined (reading 'originalname') at uploadFileToS3

S3 related code:
```js
async function uploadFileToS3(fileBuffer, fileName) {
  const params = {
    Bucket: bucketName,
    Key: fileName,
    Body: fileBuffer,
    ContentType: "application/pdf",
  };

  try {
    const response = await s3.upload(params).promise();
    return { url: response.Location, bucket: bucketName, key: fileName };
  } catch (error) {
    console.error("Error uploading file to S3:", error);
    throw new Error("Error uploading file to S3");
  }
}
```

Controller: `createProfile` of `ProfileController`.
